cmd/orch-tester/orch_tester.go

```diff
@@ -266,6 +299,68 @@ func defaultAddr(addr, defaultHost, defaultPort string) string {
 	return addr
 }
 
+func startEmbeddedStreamTester(ctx context.Context) {
+	glog.Info("Starting embedded streamtester service")
+	hostName, _ := os.Hostname()
+	streamtesterMetrics.InitCensus(hostName, model.Version, "streamtester")
+	s := server.NewStreamerServer(false, streamTesterLapiToken, streamTesterMistCreds, 4242)
+	go func() {
+		addr := fmt.Sprintf("%s:%s", "0.0.0.0", streamTesterPort)
+		s.StartWebServer(ctx, addr)
+	}()
+}
+
+func startEmbeddedBroadcaster(ctx context.Context, bCfg broadcasterConfig, presets string) {
+	glog.Info("Starting embedded broadcaster service")
+
+	// Increase Broadcaster timeouts
+	common.SegUploadTimeoutMultiplier = 4.0
+	common.SegmentUploadTimeout = 8 * time.Second
+	common.HTTPDialTimeout = 8 * time.Second
+	common.SegHttpPushTimeoutMultiplier = 4.0
+
+	// Disable caching for Orchestrator Discovery Webhook
+	common.WebhookDiscoveryRefreshInterval = 0
+
+	// Start broadcaster
+	cfg := starter.DefaultLivepeerConfig()
+	cfg.Network = bCfg.network
+	cfg.MaxSessions = intPointer(200)
+	cfg.OrchWebhookURL = stringPointer(fmt.Sprintf("http://%s:%s/orchestrators", defaultHost, streamTesterPort))
+	cfg.EthUrl = bCfg.ethUrl
+	cfg.Datadir = bCfg.datadir
+	cfg.Monitor = boolPointer(true)
+	cfg.EthPassword = bCfg.ethPassword
+	cfg.LocalVerify = boolPointer(false)
+	cfg.HttpIngest = boolPointer(true)
+	cfg.TranscodingOptions = &presets
+	cfg.MaxTicketEV = bCfg.maxTicketEV
+	cfg.MaxPricePerUnit = bCfg.maxPricePerUnit
+	cfg.CliAddr = stringPointer(fmt.Sprintf("0.0.0.0:%s", bcastCliPort))
+	cfg.Broadcaster = boolPointer(true)
+	go func() {
+		starter.StartLivepeer(ctx, cfg)
+	}()
+}
+
+func waitUntilBroadcasterIsReady(ctx context.Context, bcastHost string) {
+	rCtx, _ := context.WithTimeout(ctx, bcastReadyTimeout)
+	statusEndpoint := fmt.Sprintf("http://%s:%s/status", bcastHost, bcastCliPort)
+	ticker := time.NewTicker(200 * time.Millisecond)
+	for {
+		select {
+		case <-rCtx.Done():
+			glog.Error("Waiting for broadcaster timed out")
+			return
+		case <-ticker.C:
+			resp, err := http.Get(statusEndpoint)
```

Comment:
Maybe make sure to call `resp.Body.Close()` when you do get a response here (not really sure if its necessary, but maybe not calling it can cause a resource leak)

Reply:
Updated in cb55397a5b5d00c8754f8e15403454470013ddaa